### ISSUE TITLE
Compact mode destroys laziness

### ISSUE THREAD

**Issue by georgefst** [cdepillabout/pretty-simple#113]
Oh, and I realized that currently `pretty-simple` doesn't parse and pretty-print lazily with the compact option:

```haskell
> pPrintOpt CheckColorTty defaultOutputOptionsDarkBg { outputOptionsCompact = True }  (repeat 3)
```

This doesn't print anything.  Maybe I should make a separate issue about this...

_Originally posted by @cdepillabout in https://github.com/cdepillabout/pretty-simple/issues/84#issuecomment-1154163224_

**Comment by georgefst** [comment]
This is probably my fault. I'd never even thought to test it.

**Comment by cdepillabout** [comment]
Yeah, I think the real problem here is that `pretty-simple` doesn't have any tests checking the whole laziness thing!

**Comment by georgefst** [comment]
> Yeah, I think the real problem here is that `pretty-simple` doesn't have any tests checking the whole laziness thing!

We have at least one! I remember from https://github.com/cdepillabout/pretty-simple/pull/67#issuecomment-651701234.
